### PR TITLE
Feature/article read time

### PR DESCRIPTION
**Pull request by liruchen32**
# Issue
[[會員制] 閱讀時間-前端](https://app.asana.com/0/1203699264568808/1205988685100792/f)

# Notice

# Dependency
[npm-packages#475](https://github.com/twreporter/twreporter-npm-packages/pull/475)

### CODE REVIEW

File: src/containers/Article.js [hunk 1]
```diff
@@ -54,32 +65,115 @@ class Article extends PureComponent {
...
+    }
+  }
+
+  componentDidMount() {
+    const { fontLevel, changeFontLevel, slugToFetch, isAuthed } = this.props
+
+    if (isAuthed) {
+      window.addEventListener('scroll', this.handleScroll)
+      // Fetch the full post
+      this.fetchAFullPostWithCatch(slugToFetch)
```
Comment (Aylie-Chou) [inline src/containers/Article.js:164]: Question: why do we move `fetchAFullPostWithCatch` inside `isAuthed` rule ?

Comment (liruchen32) [reply src/containers/Article.js]: fixed

File: src/containers/Article.js [hunk 2]
```diff
@@ -54,32 +65,115 @@ class Article extends PureComponent {
...
+      activeTime: 0,
+    })
+  }
+
+  _handleVisibilityChange() {
+    if (document.visibilityState === 'hidden') {
+      if (
+        this.state.activeTime >= articleReadTimeConditionConfig.min_active_time
+      ) {
+        this.sendActiveTime()
```
Comment (Aylie-Chou) [inline src/containers/Article.js:139]: Question: should we call `calculateActiveTime` before `sendActiveTime` here ?

File: src/containers/Article.js [hunk 3]
```diff
@@ -54,32 +65,115 @@ class Article extends PureComponent {
...
+        this.setState({
+          isActive: false,
```
Comment (Aylie-Chou) [inline src/containers/Article.js:141]: Question: do we need to set isActive as true when visibilityState is `visible` ?

Comment (liruchen32) [reply src/containers/Article.js:142]: fixed

File: src/containers/Article.js (hunk 2 above)
Comment (liruchen32) [reply src/containers/Article.js:140]: fixed

**Review by Aylie-Chou — APPROVED** [review]
LGTM

**Pull request merged**